engine/src/flutter/lib/web_ui/lib/src/engine/clipboard.dart

```diff
@@ -2,240 +2,109 @@
-    MethodCodec codec,
-    dynamic error,
-  ) {
-    print('Could not get text from clipboard: $error');
-    callback!(codec.encodeErrorEnvelope(code: 'paste_fail', message: 'Clipboard.getData failed'));
-  }
-
-  /// Methods used by tests.
-  set pasteFromClipboardStrategy(PasteFromClipboardStrategy strategy) {
-    _pasteFromClipboardStrategy = strategy;
-  }
-
-  set copyToClipboardStrategy(CopyToClipboardStrategy strategy) {
-    _copyToClipboardStrategy = strategy;
-  }
-}
-
-/// Provides functionality for writing text to clipboard.
-///
-/// A concrete implementation is picked at runtime based on the available
-/// APIs and the browser.
-abstract class CopyToClipboardStrategy {
-  factory CopyToClipboardStrategy() {
-    return domWindow.navigator.clipboard != null
-        ? ClipboardAPICopyStrategy()
-        : ExecCommandCopyStrategy();
-  }
-
-  /// Places the text onto the browser Clipboard.
-  ///
-  /// Returns `true` for a successful action.
-  ///
-  /// Returns `false` for an uncessful action or when there is an excaption.
-  Future<bool> setData(String? text);
-}
-
-/// Provides functionality for reading text from clipboard.
-///
-/// A concrete implementation is picked at runtime based on the available
-/// APIs and the browser.
-abstract class PasteFromClipboardStrategy {
-  factory PasteFromClipboardStrategy() {
-    return (ui_web.browser.browserEngine == ui_web.BrowserEngine.firefox ||
-            domWindow.navigator.clipboard == null)
-        ? ExecCommandPasteStrategy()
-        : ClipboardAPIPasteStrategy();
+  /// Overrides the default clipboard strategy.
+  @visibleForTesting
+  set clipboardStrategy(ClipboardStrategy strategy) {
+    _clipboardStrategy = strategy;
   }
-
-  /// Returns text from the system Clipboard.
-  Future<String> getData();
 }
 
-/// Provides copy functionality for browsers which supports ClipboardAPI.
+/// Provides functionality for writing and reading text from the clipboard
+/// using the Clipboard API.
 ///
-/// Works on Chrome and Firefox browsers.
 /// See: https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API
-class ClipboardAPICopyStrategy implements CopyToClipboardStrategy {
-  @override
-  Future<bool> setData(String? text) async {
-    try {
-      await domWindow.navigator.clipboard!.writeText(text!);
-    } catch (error) {
-      print('copy is not successful $error');
-      return Future<bool>.value(false);
+class ClipboardStrategy {
+  /// Places the text onto the browser clipboard.
+  Future<void> setData(String? text) async {
+    final clipboard = domWindow.navigator.clipboard;
+
+    if (clipboard == null) {
+      // This can happen when the browser is outdated or the context is not
+      // secure.
+      // See: https://developer.mozilla.org/en-US/docs/Web/API/Clipboard
+      // See: https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts
+      throw StateError('Clipboard is not available in the context.');
     }
```

Comment:
Shall we move this to a getter (call it `clipboard`) and share it with `getData`?